packages/@okta/vuepress-site/use_cases/inline_hooks/registration_hook/registration_hook/index.md

```diff
@@ -0,0 +1,244 @@
+| com.okta.user.profile.update | Set or modify an attribute in the Okta user profile that will be created for this user.                 |
+| com.okta.action.update       | Allow or deny the user's registration. |
+
+To set attributes in the user's Okta profile, supply a type property set to `com.okta.user.profile.update`, together with a `value` property set to a list of key-value pairs corresponding to the Okta user profile attributes you want to set. The attributes must already exist in your user profile schema.
+
+To explicitly allow or deny registration to the user, supply a type property set to `com.okta.action.update`, together with a value property set to `{"registration": "ALLOW"}` or `{"registration": "DENY"`}. The default is to allow registration.
+
+Commands are applied in the order in which they appear in the array. Within a single `com.okta.user.profile.update` command, attributes are updated in the order in which they appear in the `value` object.
+
+You can never use a command to update the user's password, but you are allowed to set the values of attributes other than password which are designated sensitive in your Okta user schema. Note however that the values of those sensistive attributes, if included as fields in the Self-Service Registration form, are not included in the `data.userProfile` object sent to your external service by Okta. See [data.userProfile](#data-userProfile) above.
+
+#### value
+
+The `value` object is the parameter to pass to the command.
+
+For `com.okta.user.profile.update` commands, `value` should be an object containing one or more name-value pairs for the attributes you wish to update, for example:
+
+```json
+{
+   "commands":[
+      {
+         "type":"com.okta.user.profile.update",
+         "value":{
+            "middleName":"Danger",
+            "customerId":12345
+         }
+      }
+   ]
+}
+```
+
+The above example assumes that there is an attribute `customerId` defined in your Okta user schema (`middleName` is defined by default).
+
+The same result could also be accomplished by means of two separate `com.okta.user.profile.update` commands, as follows:
+
+```json
+{
+   "commands":[
+      {
+         "type":"com.okta.user.profile.update",
+         "value":{
+            "middleName":"Danger"
+         }
+      },
+      {
+         "type":"com.okta.user.profile.update",
+         "value":{
+            "customerId":12345
+         }
+      }
+   ]
+}
+```
+
+For `com.okta.action.update` commands, `value` should be an object containing the attribute `action` set to a value of either `ALLOW` or `DENY`, indicating whether the registration should be permitted or not, for example:
+
+```json
+{
+   "commands":[
+      {
+         "type":"com.okta.action.update",
+         "value":{
+            "action":"DENY",
+         }
+      }
+   ]
+}
+```
+
+Registrations are allowed by default, so setting a value of `ALLOW` for the `action` field is valid but superfluous. 
```

Comment:
nit : I almost feel we need a table for valid commands and actions, specifying defaults etc. But I'll leave it to the doc experts for the final call.

Reply:
This isn't a bad idea. I know @motikagan-okta had thoughts about consolidating the description of the `value` field for each command type. Maybe this is one way to do that. But as you say, let's defer to him.

Reply:
I will follow up after release, and see if I can come up with a better format way to organize the description of command syntax for all hook types.